pallets/asset-index/src/lib.rs

```diff
@@ -972,16 +974,16 @@ pub mod pallet {
 				Ok(())
 			})?;
 
+			// mint the given units of the SAFT asset into the treasury's account
+			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+
```

Comment:
this must happen after NAV calculation because this affects NAV.

Reply:
Great catch! moved for debugging the nav before